nimble/controller/src/ble_ll_sync.c

```diff
@@ -1009,6 +1134,21 @@ ble_ll_sync_check_acad(struct ble_ll_sync_sm *sm,
             sm->chan_map_new_instant = le16toh(chmu->instant);
             sm->flags |= BLE_LL_SYNC_SM_FLAG_NEW_CHANMAP;
             break;
+
+#if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_BIGINFO_REPORTS)
+        case BLE_LL_ACAD_BIGINFO:
+            if (ad_len - 1 == BLE_LL_SYNC_BIGINFO_LEN) {
+                ble_ll_sync_parse_biginfo(&acad[2], 0);
+                sm->flags |= BLE_LE_SYNC_SM_FLAG_BIGINFO;
+            } else if (ad_len - 1 == BLE_LL_SYNC_BIGINFO_LEN_ENC) {
+                ble_ll_sync_parse_biginfo(&acad[2], 1);
+                sm->flags |= BLE_LE_SYNC_SM_FLAG_BIGINFO;
+            } else {
+                return false;
+            }
```

Comment:
you can just pass pointer and length to function, no need to check for encryption here